components/buildless-serverless/api/v1alpha2/function_types.go

```diff
@@ -313,13 +313,22 @@ func init() {
 	SchemeBuilder.Register(&Function{}, &FunctionList{})
 }
 
+const MaxConditionMessageLength = 32768
+
+func trimConditionMessage(msg string) string {
```

Comment:
What's the reason for adding such a function? Do we update any function with a condition message longer than 32768 characters? 🤔

Reply:
Yes, we were getting the response from the azure devops repo where along with the 401 status there was an html page with base64 encoded image which was longer than 32k characters. In order to prevent that from happening I've added a custom message on authorization error + for a situation when we receive such unexpected response in some another scenario in the future - I've added this message trimming so we can see the error in the condition message/event log and not throw the condition validation error